yellowbrick/features/rankd.py

```diff
@@ -240,27 +275,173 @@ def rank(self, X, algorithm=None):
+        if is_dataframe(X):
+            X = X.as_matrix()
+
         return self.ranking_methods[algorithm](X)
 
-    def draw(self, X, **kwargs):
+    def finalize(self, **kwargs):
+        """
+        Finalize executes any subclass-specific axes finalization steps.
+        The user calls poof and poof calls finalize.
+
+        Parameters
+        ----------
+        kwargs: dict
+            generic keyword arguments
+
+        """
+        # Set the title
+        self.set_title(
+            "{} Ranking of {} Features".format(
+                self.ranking_.title(), len(self.features_)
+            )
+        )
+
+
+##########################################################################
+## Rank 1D Feature Visualizer
+##########################################################################
+
+class Rank1D(RankDBase):
+    """
+    Rank1D computes a score for each feature in the data set with
+    a specific metric or algorithm (e.g. takes the shapiro) then returns
+    them ranked as a bar plot.
+
+    Parameters
+    ----------
+    ax : matplotlib Axes, default: None
+        The axis to plot the figure on. If None is passed in the current axes
+        will be used (or generated if required).
+
+    algorithm : one of {'shapiro', }, default: 'shapiro'
+        The ranking algorithm to use, default is 'Shapiro-Wilk.
+
+    features : list
+        a list of feature names to use
+        If a DataFrame is passed to fit and features is None, feature
+        names are selected as the columns of the DataFrame.
+
+    colormap : string or cmap, default: 'RdBu_r'
+        optional string or matplotlib cmap to colorize lines
+        Use either color to colorize the lines on a per class basis or
+        colormap to color them on a continuous scale.
+
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+
+    Examples
+    --------
+
+    >>> visualizer = Rank2D()
+    >>> visualizer.fit(X, y)
+    >>> visualizer.transform(X)
+    >>> visualizer.poof()
+    """
+    ranking_methods = {
+        'shapiro': lambda X: np.array([shapiro(x)[0] for x in X.T]),
+    }
+
+    def __init__(self, ax=None, algorithm='shapiro', features=None,
+                 colormap='RdBu_r', **kwargs):
+        """
+        Initialize the class with the options required to rank and
+        order features as well as visualize the result.
+        """
+        super(Rank1D, self).__init__(ax=None, algorithm=algorithm, features=features,
+                         colormap=colormap, **kwargs)
+
+    def draw(self, R, **kwargs):
+        """
+        Draws the heatmap of the ranking matrix of variables.
+        """
+        self.ax.bar(np.arange(len(R)), R, color=self.colormap[0])
+        self.ax.set_xticks(np.arange(len(R)))
+        self.ax.xaxis.grid(False)
+        self.ax.set_xticklabels(self.features_, rotation=90, ha='left')
```

Comment:
Thanks for that context.  

I included the axis labeling in the `draw` method because to me having labels is central to making the graph interpretable, and the `finalize` method is not called by the "quick" methods.  So, if labels are not added in `draw`, this makes the quick methods much less useful IMO. 

If the user does not provide feature names, the class is already set up to label the ticks with integers.  I should have included a demo of this, but it's already taken care of. 

On your second point about cases with many features, I agree that's a bit of a hairy issue but the approach you proposed should work for now.  I think it's useful to have the ability to put in the labels even if you have to turn them off for high-dimensional data.